subprojects/file-watching/src/main/java/org/gradle/internal/watch/registry/impl/WatchableHierarchies.java

```diff
@@ -88,6 +90,7 @@ public void registerWatchableHierarchy(File watchableHierarchy, SnapshotHierarch
         }
         if (unwatchableFiles.contains(watchableHierarchyPath)) {
             LOGGER.info("Not watching {} since the file system is not supported", watchableHierarchy);
+            ignoredUnwatchableHierarchiesSinceLastBuildFinish.add(watchableHierarchy);
```

Comment:
I feel uneasy about this. We will later call back here with the contents of this list, but then rely on `unwatchableFiles` being empty.

I think at least we should move the actual registering (without the checks for `unwatchableFiles`) to a separate method, like `registerWatchableHierarchyInternal()`.

But what potentially sounds like a better option for me is to address the problem from a different angle. I haven't thought this through, but can't we just record registered hierarchies at this point, and then handle them all at build start? It seems like a simpler process altogether than making a special case for stuff ignored between builds.

Reply:
I extracted a method and simplified the process.